### PR TITLE
Don't look for agda-lib files in /nix/store

### PR DESCRIPTION
**Pull request by ncfavier**
I'm using Agda on NixOS. When messing around trying to build some agda project with Nix, I ended up adding its `agda-lib` file to the Nix store, i.e. I had a file `/nix/store/79zjax75w4fs9az22m8i29k1ym6ja2n6-1lab.agda-lib`. As a consequence, whenever Agda tried to import a file in its built-in library (located at `/nix/store/b1hqd3g7xn8a36bq6a0bsl746qc3i9yx-Agda-2.6.2.2-data/share/ghc-9.0.2/x86_64-linux-ghc-9.0.2/Agda-2.6.2.2/lib/prim/Agda`), it would explore the filesystem upwards looking for an `agda-lib` file, which would normally fail *but now succeeded* because it found `79zjax75w4fs9az22m8i29k1ym6ja2n6-1lab.agda-lib` under `/nix/store`! This is very wrong, and caused Agda to fail immediately because that file contained a flag that's only understood by Agda 2.6.3.

This patch hardcodes `/nix/store` as a "ceiling" so that we don't ever try to look for library files in `/nix/store`. This is a bit ugly, but I don't have any other ideas.

~Also updates nixpkgs (which unbreaks vector-hashtables) and adds cabal to the development shell.~

### CODE REVIEW

File: src/full/Agda/Interaction/Library.hs [hunk 1]
```diff
@@ -181,21 +181,25 @@ findProjectConfig'
...
-  getCachedProjectConfig root >>= \case
-    Just conf -> return conf
-    Nothing   -> do
-      libFiles <- liftIO $ filter ((== ".agda-lib") . takeExtension) <$> getDirectoryContents root
-      case libFiles of
-        []     -> liftIO (upPath root) >>= \case
-          Just up -> do
-            conf <- findProjectConfig' up
+  root <- liftIO (canonicalizePath root)
+  if root == "/nix/store"
```
Comment (andreasabel) [inline src/full/Agda/Interaction/Library.hs:191]: Is this absolute path OS-agnostic?

**Comment by andreasabel** [comment]
@mergifyio rebase

**Comment by andreasabel** [comment]
Rebasing this on master...  This might fix the CI failure.

**Comment by nad** [comment]
Related: #4526. Some suggestions from that ticket that weren't implemented:
* Add a `.agda-lib` file for the builtin modules.
* Only allow a single, fixed name for `.agda-lib` files.

File: src/full/Agda/Interaction/Library.hs (hunk 1 above)
Comment (ncfavier) [reply src/full/Agda/Interaction/Library.hs:191]: Nix doesn't support any OSes that use a different separator than `/`, but it does support running in a custom prefix, e.g. `~/.nix/store`. There's no hope to cover all such possible configurations, so I think this is good enough.

**Comment by ncfavier** [comment]
> Related: #4526. Some suggestions from that ticket that weren't implemented:
> 
>  * Add a `.agda-lib` file for the builtin modules.
> 
>  * Only allow a single, fixed name for `.agda-lib` files.

If we implement either of these I think we can drop this PR.

~I like the idea of only looking for `dir/dir.agda-lib` instead of `dir/*.agda-lib`.~ Never mind, this would make Agda even less usable with Nix (we wouldn't find `/nix/store/0000000000000-cubical/cubical.agda-lib` for example).

**Comment by nad** [comment]
> ~I like the idea of only looking for `dir/dir.agda-lib` instead of `dir/*.agda-lib`.~ Never mind, this would make Agda even less usable with Nix (we wouldn't find `/nix/store/0000000000000-cubical/cubical.agda-lib` for example).

I think the idea was to use the same name for all libraries.

**Comment by ncfavier** [comment]
That definitely sounds more sensible!

**Comment by andreasabel** [comment]
> Related: #4526. Some suggestions from that ticket that weren't implemented:
> 
> * Add a `.agda-lib` file for the builtin modules.

This could likely be done right away.

> * Only allow a single, fixed name for `.agda-lib` files.

This would need a deprecation period.
I always use `Main.agda-lib`, but `package.agda-lib` would be more canonical (cf. `package.yaml` for `hpack` /`stack` or `package.json` for JS).
However, it seems that for published packages a given name like `standard-library.agda-lib` makes some sense.

**Comment by nad** [comment]
> However, it seems that for published packages a given name like `standard-library.agda-lib` makes some sense.

The point of having a fixed name is to avoid having to check if one of very many files in a directory is a `.agda-lib` file.

**Comment by MatthiasHu** [comment]
> * Add a `.agda-lib` file for the builtin modules.

This was previously attempted by @gallais here: https://github.com/agda/agda/pull/4616

**Comment by ncfavier** [comment]
I'd like to either close or merge this.

**Comment by andreasabel** [comment]
Sorry, @ncfavier, for the lack of response.  I guess hardcoding a path didn't find much love, and the discussion derailed a bit.

**Pull request closed without merging**